examples/wakaama/README.md

```diff
@@ -0,0 +1,103 @@
+- [Server object](http://www.openmobilealliance.org/tech/profiles/LWM2M_Server-v1_0.xml)
+- [Access control object](http://www.openmobilealliance.org/tech/profiles/LWM2M_Access_Control-v1_0_2.xml)
+- [Device object](http://www.openmobilealliance.org/tech/profiles/LWM2M_Device-v1_0_3.xml)
+- [Light control object](http://www.openmobilealliance.org/tech/profiles/lwm2m/3311.xml)
+
+It will create an instance of the Light control object per LED (LED0 and LED1)
+if available. The state of the LED can be changed from the LWM2M server or by
+using a shell command in which case the state of it will be reflected on the
+LWM2M server.
+
+The application is based on the Eclipse Wakaama
+[example client](https://github.com/eclipse/wakaama/tree/master/examples/client)
+.
+
+## Usage
+
+### Setting up a LWM2M Test Server
+To test the client a LWM2M server where to register is needed.
+[Eclipse Leshan](https://github.com/eclipse/leshan) demo is a good option for
+running one locally.
+
+To run the demo server:
+```shell
+wget https://hudson.eclipse.org/leshan/job/leshan/lastSuccessfulBuild/artifact/leshan-server-demo.jar
+
+java -jar ./leshan-server-demo.jar
+```
+It will output the addresses where it is listening:
+```
+INFO LeshanServer - LWM2M server started at coap://0.0.0.0/0.0.0.0:5683 coaps://0.0.0.0/0.0.0.0:5684
+INFO LeshanServerDemo - Web server started at http://0.0.0.0:8080/.
+```
+
+#### Bootstrap server
+LWM2M provides a bootstrapping mechanism to provide the clients with information
+to register to one or more servers. To test this mechanism both the previous server and a bootstrap server should be running. Eclipse Leshan also provides a bootstrap server demo.
+
+By default the bootstrap server option is disabled, it can be enabled by setting
+`LWM2M_SERVER_IS_BOOTSTRAP` in `sys/net/lwm2m.h` to 1.
+
+To run the bootstrap server, make sure that the ports it uses are different
+from the ones of previous server (default are 5683 for CoAP, 5684 for CoAPs,
+and 8080 for the webserver), and that it corresponds to the one set in
+`sys/net/lwm2m.h` as `LWM2M_BSSERVER_PORT`:
+```shell
+# download demo
+wget https://hudson.eclipse.org/leshan/job/leshan/lastSuccessfulBuild/artifact/leshan-bsserver-demo.jar
+
+# set CoAP, CoAPs and webserver ports for bootstrap server
+BS_COAPPORT=5685
+BS_COAPSPORT=5686
+BS_WEBPORT=8888
+
+# run the server
+java -jar ./leshan-bsserver-demo.jar --coapport ${BS_COAPPORT} \
+            --coapsport ${BS_COAPSPORT} --webport ${BS_WEBPORT}
+```
+
+To set up the configuration of the node and the server:
+1. Click the `Add new client bootstrap configuration` button.
+2. Fill in the name of the device, it **should** match the one set in
+   `sys/net/lwm2m.h` as `LWM2M_DEVICE_NAME`.
+3. Using the `LWM2M Server` tab enter the address where the LWM2M server is
+   listening. For now only `No security` mode can be used.
+
+### Running the client
+The address set in `sys/net/lwm2m.h` as `LWM2M_SERVER_URI` should be reachable
```

Comment:
I found that I had to include the port in LWM2M_SERVER_URI (actually SERVER_URI in the Makefile) to get this to work with a bootstrap server on port 5685. For example: `SERVER_URI="coap://[fd00:bbbb::1]:5685"` In other words, it is not enough to have LWM2M_BSSERVER_PORT set to 5685. Did it work for you without including the port in LWM2M_SERVER_URI?

Reply:
You are right, it was not being used, I fixed that now.